internal/postgresql/postgresql.go

```diff
@@ -222,6 +224,13 @@ func (p *Manager) Init(initConfig *InitConfig) error {
 	}
 	log.Debugw("execing cmd", "cmd", cmd)
 
+	// initdb supports configuring a separate wal directory via symlinks. Normally this
+	// parameter might be part of the initConfig, but it will also be required whenever we
+	// fall-back to a pg_basebackup during a re-sync, which is why it's a Manager field.
+	if p.walDir != "" {
+		cmd.Args = append(cmd.Args, "--waldir", p.walDir)
+	}
+
```

Comment:
@lawrencejones probably I'm missing something but I don't see the code that recreates the pg_xlog/pg_wal symlink to the real data dir inside the postgres data. I.E. suppose that the wal_dir si different between the keepers and/or that its value has changed since the user moved the wal_dir data to another path, during instance start we should recreate the symlink.

Reply:
@sgotti , all of this is handled by pg_basebackup.
If you run pg_basebackup with the --waldir option, it will create that folder, copy the wal data there and create the symlink.
It is transparent to Postgres what the symlink points to from where you are streaming from.
Like if master has pg_wal a folder in PGDATA, and you run pg_basebackup with --waldir option set to a folder outside of PGDATA, the new PGDATA would have the symlink correctly setup.